src/components/nav/nav.tsx

```diff
@@ -0,0 +1,198 @@
+
+  /**
+   * When `true`, displays a `calcite-action` menu and emits a `calciteNavActionSelect` event on selection change.
+   */
+  @Prop({ reflect: true }) navAction = false;
+
+  /**
+   * Specifies the label of the hamburger icon.
+   */
+  @Prop() label!: string;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Properties/ State
+  //
+  //--------------------------------------------------------------------------
+
+  @State() logoSlotHasElements: boolean;
+
+  @State() userSlotHasElements: boolean;
+
+  @State() primarySlotHasElements: boolean;
+
+  @State() secondarySlotHasElements: boolean;
+
+  @State() teritiarySlotHasElements: boolean;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /** When navAction is true, emits when the calcite-action menu selection changes.*/
+  @Event() calciteNavActionSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clickHandler = () => {
+    this.calciteNavActionSelect.emit();
+  };
+
+  private handleContentSlotChange = (event: Event, level: Level): void => {
+    if (level === LEVEL.secondary) {
+      this.secondarySlotHasElements = slotChangeHasAssignedElement(event);
+    } else if (level === LEVEL.tertiary) {
+      this.teritiarySlotHasElements = slotChangeHasAssignedElement(event);
+    } else {
+      this.primarySlotHasElements = slotChangeHasAssignedElement(event);
+    }
+  };
+
+  private handleUserSlotChange = (event: Event): void => {
+    this.userSlotHasElements = slotChangeHasAssignedElement(event);
+  };
+
+  private handleLogoSlotChange = (event: Event): void => {
+    this.logoSlotHasElements = slotChangeHasAssignedElement(event);
+  };
+
+  private handleMenuActionSlotChange = (event: Event): void => {
+    const hasMenuAction = slotChangeHasAssignedElement(event);
+    if (hasMenuAction) {
+      this.navAction = false;
+    }
+  };
+
+  private hasSlottedElements(level: Level): boolean {
+    if (level === LEVEL.primary) {
+      return (
+        this.navAction ||
+        this.userSlotHasElements ||
+        this.logoSlotHasElements ||
+        this.primarySlotHasElements
+      );
+    } else if (level === LEVEL.secondary) {
+      return this.secondarySlotHasElements;
+    } else {
+      return this.teritiarySlotHasElements;
+    }
+  }
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderMenuAction(): VNode {
+    return (
+      <slot name={SLOTS.navAction} onSlotchange={this.handleMenuActionSlotChange}>
+        {this.navAction && (
```

Comment:
I don't think you need the check for `this.navAction` because this is fallback content by default. If a user adds this slot then the content inside it won't render.

Reply:
didn't knew that. Will update this.